qpc/insights/utils.py

```diff
@@ -126,3 +128,15 @@ def format_upload_success(streamdata):
         return stream_list[2]
     except IndexError:
         return streamdata
+
+
+def validate_host(arg):
+    """Check that arg is a valid host.
+
+    :param arg: a string
+    :returns: the arg
+    :raises: ArgumentTypeError, if arg is an empty string, None or not a string
+    """
+    if isinstance(arg, str) and arg != "":
+        return arg
+    raise ArgumentTypeError(f"Host value {arg} should be a valid non-empty string")
```

Comment:
Nice!

@nicolearagao, I'm assuming ArgumentTypeError will be displayed as a nice error (instead of an ugly traceback) for the end user. Is this understanding correct?

Reply:
If you follow gregg suggestion (i think its a great idea) make sure to rename this function/docstring to reflect this behavior